config/param-validation.js

```diff
@@ -1,14 +1,22 @@
 import Joi from 'joi';
 
 export default {
-  // UPDATE /api/users/:userId
+  // UPDATE /api/event
   event: {
     body: {
       clientId: Joi.string().required(),
-      deviceType: Joi.string().required(),
+      deviceType: Joi.string().required().valid(['iOS', 'Android', 'Browser']),
       location: Joi.string().required(),
-      eventTime: Joi.string().required(),
-      eventType: Joi.string().required()
+      eventTime: Joi.date().timestamp('unix').required(),
+      eventType: Joi.string().required().valid(
+        [
+          'login',
+          'logout',
+          'play_episode',
+          'pause_episode',
+          'like_episode',
+          'completed_episode'
+        ])
```

Comment:
@andrewmarklloyd  I think We would need to add another attribute, ```eventData``` to our  ```event``` object. ```EventData``` will be an object containing metadata  specific to the event being fired.  This metadata will have  its own schema which we would need to validate.  
For example, when say a ```pause_episode``` event is fired, the  ```eventData``` attribute  of our ```event``` object  would be an object with attributes such  as  ```minutes_remaining``` , ```minutes_playes```, ```episode_name``` etc.  Following is a crude example
```javascript
event: {
      body: {
        clientId: Joi.string().required(),
       deviceType: Joi.string().required(),
       deviceType: Joi.string().required().valid(['iOS', 'Android', 'Browser']),
        location: Joi.string().required(),
       eventTime: Joi.string().required(),
       eventType: Joi.string().required()
       eventTime: Joi.date().timestamp('unix').required(),
       eventType: Joi.string().required().valid(
         [
           'login',
           'logout',
           'play_episode',
           'pause_episode',
           'like_episode',
           'completed_episode'
        ],
       eventData: Joi.object()
)

pauseEpisodeEvent: {
episodeName: Joi.string().required(),
minutesPlayed: Joi.number().required(),
minutesRemaining: Joi.number().required()
}

```
With this, we will end up doing a two a step validation: (1) Validate the top level event  schema (2) Based on the eventType, validate the eventData schema. 

What are your thoughts?  Perhaps there is a way to avoid the two step validation? .

We can handle this in a separate github issue which I will create